### PR TITLE
fix(Script/Mother Shahraz) Add new teleport points.

### PR DESCRIPTION
**Pull request by Rorschach91**
This fix adds 5 new teleport points if the boss is near the platform.

## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [X] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/20135#issue-2567813712

## SOURCE:

The changes have been validated through:
- [X] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [X] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [X] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [X] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. Tank the boss around the platform
2. Tank the boss around the waterfall
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Outland/BlackTemple/boss_mother_shahraz.cpp [hunk 1]
```diff
@@ -196,15 +196,25 @@ class spell_mother_shahraz_saber_lash_aura : public AuraScript
...
     {927.22f, 187.04f, 192.84f},
     {922.54f, 110.09f, 192.84f},
-    {958.01f, 110.47f, 192.84f}
+    {958.01f, 110.47f, 192.84f},
+    {939.95f, 108.29f, 192.84f},
+    {945.68f, 205.74f, 192.84f}
 };
 
-constexpr float minTeleportDist = 30.f;
+constexpr float minTeleportDist = 35.f;
```
Comment (heyitsbench) [inline src/server/scripts/Outland/BlackTemple/boss_mother_shahraz.cpp:217]: This is wrong. Comparing positions of one teleport with a sniff yields the following:
`The distance is: (3D) 30.496243 (2D) 30.202114 (Exact 3D) 32.996243 (Exact 2D) 32.702114 yards.`

Comment (Rorschach91) [reply src/server/scripts/Outland/BlackTemple/boss_mother_shahraz.cpp:217]: If exact 2D/3D are around 32 yards (almost 33) should I modify to 32f?
Or Should I return to 30f?

Comment (heyitsbench) [reply src/server/scripts/Outland/BlackTemple/boss_mother_shahraz.cpp:217]: I would say return it to 30f.

Comment (Rorschach91) [reply src/server/scripts/Outland/BlackTemple/boss_mother_shahraz.cpp:217]: Roger.

**Review by Helias — APPROVED** [review]

**Pull request merged**